packages/counterfactual_nft/contracts/NFTFactory.sol

```diff
@@ -44,9 +46,10 @@ contract NFTFactory
         );
 
         // Initialize
+        IOpenseaSupport(nftContract).setContractURI(contractURI);
         ICounterfactualNFT(nftContract).initialize(owner, baseURI);
```

Comment:
Maybe better to add `contractURI` to `initialize` as well so all values are set together?

Reply:
Because I don't want to change the behavior of CounterfactualNFT, I added the field contractURI to OpenseaSupport contract, otherwise I will have to add it to CounterfactualNFT

Reply:
Fair enough, though you could also just override `initialize` in `CounterfactualNftExt` which may still be a bit cleaner so there's no special logic with the owner in `setContractURI`. But I'm okay with how it is now. If you keep the different calls a small gas cost improvement is probably to only call `setContractURI` if `contractURI.length > 0`.